pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -460,3 +614,14 @@ func TestFullMonitorWithTracer(t *testing.T) {
 
 	initTracerState(t, tr)
 }
+
+func attachPID(t *testing.T, pid int) {
+	t.Helper()
+	require.NoError(t, usm.GoTLSAttachPID(uint32(pid)))
+	utils.WaitForProgramsToBeTraced(t, "go-tls", pid)
+}
+
+func detachPID(t *testing.T, pid int) {
+	t.Helper()
+	require.NoError(t, usm.GoTLSDetachPID(uint32(pid)))
+}
```

Comment:
nit: maybe check the program is traced before calling detach (it is not, then silently abort)
and after the detach, check the program is not traced